src/ResourceManager.cpp

```diff
@@ -0,0 +1,46 @@
+#include "ResourceManager.h"
+
+ResourceManager::ResourceManager(){}
+ResourceManager::~ResourceManager(){}
+
+void ResourceManager::LoadResources() {
+	LoadTexture("Duck", "Res/JanitorDuck.png");
+	LoadTexture("Meadow", "Res/quackMeadowBG.png");
+
+	LoadSound("Quack", "Res/quack.wav");
+	LoadSound("Frog", "Res/froggy.wav");
+	LoadSound("Catch", "Res/frogCatch.wav");
+
+	LoadFont("Text", "Res/consola.ttf");
+}
+
+void ResourceManager::LoadTexture(const char* name, const char* path) {
+    sf::Texture& tex = TextureMap[name]; 
+	tex.loadFromFile(path);
+};
+
+void ResourceManager::LoadSound(const char* name, const char* path) {
+	sf::SoundBuffer& Buffer = BufferMap[name];
+	Buffer.loadFromFile(path);
+	
+	sf::Sound& Sound = SoundMap[name];
```

Comment:
we don't need a map for sf::Sound - it's not a resource.

sf::Sound is an audio version of sf::Sprite - it only refers to an actual resource held by sf::SoundBuffer.

So just like with sprites, any class that wants to have audio has a SoundComponent as it's member, and a SoundComponent is just an sf::Sound pointing to the buffer held by the ResourceManager